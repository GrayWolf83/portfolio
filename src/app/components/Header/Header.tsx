import React from 'react'
import { AppBar, Container, Toolbar, Typography } from '@mui/material'
import { Link } from 'react-router-dom'

const Header: React.FC = () => {
	return (
		<AppBar position='static'>
			<Container>
				<Toolbar>
					<Link to='/'>
						<Typography
							variant='h5'
							component='h2'
							sx={{
								flexGrow: 1,
								letterSpacing: 1.5,
							}}>
							GitHub view
						</Typography>
					</Link>
				</Toolbar>
			</Container>
		</AppBar>
	)
}

export default Header
