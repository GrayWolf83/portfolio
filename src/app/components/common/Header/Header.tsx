import React from 'react'
import { AppBar, IconButton, Toolbar, Typography } from '@mui/material'
import { Info } from '@mui/icons-material'
import { Link } from 'react-router-dom'

const Header = () => {
	return (
		<AppBar position='static'>
			<Toolbar>
				<Link to='/'>
					<Typography
						variant='h5'
						component='h2'
						sx={{
							flexGrow: 1,
							letterSpacing: 1.5,
						}}>
						GitHub view
					</Typography>
				</Link>
			</Toolbar>
		</AppBar>
	)
}

export default Header
