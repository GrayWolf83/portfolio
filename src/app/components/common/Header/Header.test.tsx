import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import Header from './Header'

describe('Header component', () => {
	it('Header render', () => {
		render(
			<BrowserRouter>
				<Header />
			</BrowserRouter>,
		)

		expect(screen.getByText(/github/i)).toBeInTheDocument()
	})

	it('Header snapshot', () => {
		expect(
			render(
				<BrowserRouter>
					<Header />
				</BrowserRouter>,
			),
		).toMatchSnapshot()
	})
})
