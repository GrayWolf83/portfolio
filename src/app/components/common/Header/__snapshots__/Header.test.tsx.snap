// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <a
            href="/"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 css-cntv86-MuiTypography-root"
            >
              GitHub view
            </h2>
          </a>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <a
          href="/"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 css-cntv86-MuiTypography-root"
          >
            GitHub view
          </h2>
        </a>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
